### PR TITLE
Customize the site navbar and masthead

### LINKED ISSUE #937: Customize site masthead structure and styling, Version 2
Replaces #884. That ticket has a long description and it seems easier to start with a fresh one.

The main goal is to customize the site masthead structure; I'll provide some styling suggestions but after the structure is in place, I can refine colors and font-sizes as necessary in a follow-on PR.

As an overview, we're sticking with the current default Spotlight main menu and search box bar. The customizations we need are:

- Make the topbar taller to accommodate the DLME logo and site title
- Add the DLME logo to the topbar
- Move the site title `H1` into the topbar
- Enable the masthead background image to extend up so it is underneath the entire masthead section, including the topbar (so we can have it show through the semi-transparent topbar)

## LG and XL viewports
Screen Shot 2020-04-08 at 3 52 25 PM

### Masthead height and background
- The total height (including topbar section) should be about `178px`, but it can be determined by the heights of the individual sections.

- The masthead background image: `background-light-blue-mosaic.png` 

### Masthead topbar
- Topbar height:  should end up about`66px`
- Topbar background color: `#0F2450`
- Topbar font color: `#ffffff`

- Logo image: `DLME-logo-transparent-white.png` at `50px` height. The logo is not linked.
- Logo image top and bottom margin: `8px`
- Logo right margin: to get the horizontal space between the logo and the site title, I did:

```CSS
.topbar .navbar-logo {
  flex: 0 0 80px; // was 0 0 150px
}
```

- Site title (`Digital Library of the Middle East`): `2rem`
- Site title vertical placement: centered with the logo; I gave it `margin-top: 1rem;` but there is a probably a better way to ensure vertical centering in the topbar
- Topbar utility links: `1rem` (same as current), `#ffffff` (changed)

### Masthead body (menu and searchbox section)
Everything is standard Spotlight, except in my mockup I'm showing the active menu item with a blue (`#0F2450`) background behind it to indicate that it's linked. This is how we did the main menu before the BL7/BS4 upgrade but we lost that because the `.navbar` element now has top and bottom padding, which prevents the highlight background from extended the full height of the menu bar, so it looks weird. We can try to stick with depending on the font color change of the active menu item to indicate it is active, but if we can figure out how to show the active item like in the mockup, that would be better.

### Body background color

`body` background color: Change to `#f7f7f7`.

## Responsive considerations
Below are responsive adjustments needed when the viewport is below the `LG` (`992px`) viewport width. These are general suggestions; feel free to do things differently when it makes sense. The goal is to approximate the screenshots and I can tweak later.

### MD viewport
Screen Shot 2020-04-08 at 4 28 20 PM

- Site title: Reduce font-size to `1.75rem`
- Site title top-margin: Reduce to `0.5rem`

### SM viewport
Screen Shot 2020-04-08 at 4 33 20 PM

- Site title: Reduce font-size to `1.5rem`
- Site title top-margin: Reduce to `0.5rem` (same as MD)
- Reduce logo size and right-margin:

```CSS
.topbar .navbar-logo {
  flex: 0 0 60px; 
}

.navbar-brand {
  margin-right: 0rem;
}
```

### XS viewport
Screen Shot 2020-04-08 at 4 40 12 PM

- Site title: Reduce font-size to `1.25rem`
- Site title top-margin: Reduce to `0.5rem` (same as MD and SM)
- Logo: `display: none`

## Logo image file
DLME-logo-transparent-white

## Masthead background image file
background-light-blue-mosaic

### PR DESCRIPTION
**Pull request by cbeer**
Screen Shot 2020-04-09 at 07 49 40
Fixes #937

Made a little better by:
- [x] https://github.com/projectblacklight/spotlight/pull/2517 (our masthead aspect ratio is a little different than out-of-the-box spotlight)
- [x] https://github.com/projectblacklight/spotlight/pull/2518  (stop overriding the override of  the override)

Open questions:
- [x] Do we need to do anything special for browse categories with  mastheads? [No, DLME isn't  going to use that option]
- [ ] Is there an SVG version of the logo (and, should we also use  it for a favicon?)
- [x] #937 says the site bar should have transparency, but no value was specified.
- [x] Does the hover/focus  state styling for the site navbar need to change too?

### CODE REVIEW

**Comment by ggeisler** [comment]
> Do we need to do anything special for browse categories with mastheads?

Nope

> Is there an SVG version of the logo (and, should we also use it for a favicon?)

Working on getting an SVG version; I think it makes sense to use it for a favicon, good idea

> #937 says the site bar should have transparency, but no value was specified.

I used 75% in my mockup, so we can start there.

> Does the hover/focus state styling for the site navbar need to change too?

Not sure which elements you're referring to. The utility links (language, user login)? If so, I guess I was assuming we'd inherit what we did in the BL7/BS4 update: showing `text-decoration: underline` on hover but no color change to the links.

**Comment by cbeer** [comment]
> I guess I was assuming we'd inherit what we did in the BL7/BS4 update:

From exhibits?

File: app/assets/stylesheets/variables.scss [hunk 1]
```diff
@@ -18,9 +18,9 @@ $pagination-active-color: $color-blackish;
...
 $pagination-active-border: $color-gray-medium;
 
-$body-bg: $color-gray-lightest;
-
 $breadcrumb-bg: transparent;
 $breadcrumb-active-color: $color-black-medium;
 $breadcrumb-divider-color: $color-black-light;
 $breadcrumb-padding-x: 0.5rem;
+
+$topnav-background-color: rgba(#0f2450, 0.9);
```
Comment (cbeer) [inline app/assets/stylesheets/variables.scss:26]:
```suggestion
$topnav-background-color: rgba(#0f2450, 0.75);
```

**Comment by ggeisler** [comment]
Yes, from Exhibits (I don't have an easy way to look at Spotlight without spinning it up locally so wasn't sure what we do there).

**Pull request merged**